File: game/forest/hurt_trap_animation.py
from animation import Animation
from forest.forest_resources import ForestResources
from forest.hurt_trap_talking import HurtTrapTalking
from state import State


class HurtTrapAnimation(State):
    def process_events(self, events):
        if self.get_frame_index() >= len(ForestResources.hugo_traphurt):
            return HurtTrapTalking(self.parent)

    def render(self, screen):
        screen.blit(Animation.get_frame(ForestResources.hugo_traphurt, self.get_frame_index()), (0,0))

File: game-web/src/forest/hurtTrapAnimation.ts
/** Port of game/forest/hurt_trap_animation.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { ForestResources } from './forestResources';
import { HurtTrapTalking } from './hurtTrapTalking';
import type { PhoneEvents } from '../phoneEvents';

export class HurtTrapAnimation extends State {
  processEvents(_events: PhoneEvents): StateClass | null {
    return this.getFrameIndex() >= ForestResources.hugoTraphurt.length ? HurtTrapTalking : null;
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getFrame(ForestResources.hugoTraphurt, this.getFrameIndex()), 0, 0);
  }
}
